### ISSUE TITLE
How to use op_thresh during model testing

### ISSUE THREAD

**Issue by adwaykanhere** [mlmed/torchxrayvision#88]
Hi,

I'm using the library to measure cross-domain generalization performance using the pre-trained models. For example, I'm using the Chexpert pre-trained model and testing it on another dataset to measure it's performance - I would like to know how I can use the op_thresh attribute of the model to threshold the outputs for metric calculation (F1, Sensitivity, etc.). Or should I use 0.5 as the threshold?

I'm new to using the library and any suggestions are greatly appreciated. Thanks!

**Comment by ieee8023** [comment]
The models are calibrated on their respective validation sets so that 0.5 is the decision boundary so yes you could just threshold the output at 0.5.  

However, these values may not be perfect which is just a challenge of model generalization. We discuss this more in section 3 here: https://arxiv.org/abs/2002.02497 more research is needed here to know the best approach to address this problem. We just went with that specific approach.

**Comment by adwaykanhere** [comment]
Thank you! Another point I would like to clarify, what value do I pass to op_thresh when calling the model? For example, I set apply_sigmoid = True but can't do the same for op_thresh. Is it being done internally?

**Comment by ieee8023** [comment]
`op_threshs` is being set internally when you specify weights. What you can do to disable them is to load the model and then set the op_threshs to None like this:

```python
model = xrv.models.DenseNet(weights="densenet121-res224-all")
model.op_threshs = None
```

This will prevent them from being applied here: https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/models.py#L251

This is highlighting how this interface is not the best and the argument `op_threshs` is ignored in the model constructor. Maybe an argument apply_op_threshs would be a more clear interface?

**Comment by adwaykanhere** [comment]
Yes, another argument for using the thresholds would be a great help. Thank you for the clarification. 
I will close this issue now.
